src/main/java/htmlpublisher/workflow/WorkflowActionsFactory.java

```diff
@@ -0,0 +1,67 @@
+/*
+ * The MIT License
+ *
+ * Copyright 2015 CloudBees Inc.
+ *
+ * Permission is hereby granted, free of charge, to any person obtaining a copy
+ * of this software and associated documentation files (the "Software"), to deal
+ * in the Software without restriction, including without limitation the rights
+ * to use, copy, modify, merge, publish, distribute, sublicense, and/or sell
+ * copies of the Software, and to permit persons to whom the Software is
+ * furnished to do so, subject to the following conditions:
+ *
+ * The above copyright notice and this permission notice shall be included in
+ * all copies or substantial portions of the Software.
+ *
+ * THE SOFTWARE IS PROVIDED "AS IS", WITHOUT WARRANTY OF ANY KIND, EXPRESS OR
+ * IMPLIED, INCLUDING BUT NOT LIMITED TO THE WARRANTIES OF MERCHANTABILITY,
+ * FITNESS FOR A PARTICULAR PURPOSE AND NONINFRINGEMENT. IN NO EVENT SHALL THE
+ * AUTHORS OR COPYRIGHT HOLDERS BE LIABLE FOR ANY CLAIM, DAMAGES OR OTHER
+ * LIABILITY, WHETHER IN AN ACTION OF CONTRACT, TORT OR OTHERWISE, ARISING FROM,
+ * OUT OF OR IN CONNECTION WITH THE SOFTWARE OR THE USE OR OTHER DEALINGS IN
+ * THE SOFTWARE.
+ */
+package htmlpublisher.workflow;
+
+import htmlpublisher.HtmlPublisherTarget;
+import hudson.Extension;
+import hudson.model.Action;
+import hudson.model.Job;
+import hudson.model.Run;
+import java.util.Collection;
+import java.util.LinkedList;
+import java.util.List;
+import jenkins.model.TransientActionFactory;
+import org.kohsuke.accmod.Restricted;
+import org.kohsuke.accmod.restrictions.DoNotUse;
+
+/**
+ * Produces actions for workflow jobs.
+ * @author Oleg Nenashev
+ * @since TODO
+ */
+@Restricted(DoNotUse.class)
+@Extension
+public class WorkflowActionsFactory extends TransientActionFactory<Job> {
```

Comment:
I've started from this approach, but reverted changes after getting into troubles with dependencies. SimpleBuildStep.LastBuildAction may present in SimpleBuildStep only, but the migration of the publisher to this would require a refactoring of all initial classes (HTMLPublisher target should become describable, etc.). Multi-entry selection in a single Workflow step is not a thing I would like to have there.